linera-views/src/reentrant_collection_view.rs

```diff
@@ -491,19 +515,22 @@ where
                 .base_tag_index(KeyTag::Subview as u8, short_key);
             let context = self.context.clone_with_base_key(key);
             match updates.entry(short_key.to_vec()) {
-                btree_map::Entry::Occupied(entry) => {
-                    let entry = entry.into_mut();
-                    if let Update::Removed = entry {
+                btree_map::Entry::Occupied(mut entry) => {
+                    if let Update::Removed = entry.get() {
                         let view = W::new(context)?;
                         let view = Arc::new(RwLock::new(view));
-                        *entry = Update::Set(view);
+                        entry.insert(Update::Set(view));
+                        cached_entries.remove(short_key);
                     }
                 }
                 btree_map::Entry::Vacant(entry) => {
                     if self.delete_storage_first {
+                        cached_entries.remove(short_key);
                         let view = W::new(context)?;
                         let view = Arc::new(RwLock::new(view));
                         entry.insert(Update::Set(view));
+                    } else if let Some(view) = cached_entries.remove(short_key) {
+                        entry.insert(Update::Set(view));
                     } else {
                         keys.extend(W::pre_load(&context)?);
                         short_keys_to_load.push(short_key.to_vec());
```

Comment:
Perhaps the code below could be made closer to the new `try_load_entries`

Reply:
I think I can refactor this to not need the manual index calculations. What do you think of [this](https://github.com/jvff/linera-protocol/commit/d80fea02a099f0b1728afb57d54e9d3cad8bf8ae)? I can open a PR with that change after this one is merged :+1:

Reply:
Nice!